pkg/artifactory/resource/configuration/resource_artifactory_mail_server.go

```diff
@@ -0,0 +1,338 @@
+		Attributes: map[string]schema.Attribute{
+			"enabled": schema.BoolAttribute{
+				MarkdownDescription: "When set, mail notifications are enabled.",
+				Required:            true,
+			},
+			"artifactory_url": schema.StringAttribute{
+				MarkdownDescription: "The Artifactory URL to to link to in all outgoing messages.",
+				Optional:            true,
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(1),
+					validatorfw_string.IsURLHttpOrHttps(),
+				},
+			},
+			"from": schema.StringAttribute{
+				MarkdownDescription: "The 'from' address header to use in all outgoing messages.",
+				Optional:            true,
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(1),
+					validatorfw_string.IsEmail(),
+				},
+			},
+			"host": schema.StringAttribute{
+				MarkdownDescription: "The mail server IP address / DNS.",
+				Required:            true,
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(1),
+				},
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"username": schema.StringAttribute{
+				MarkdownDescription: "The username for authentication with the mail server.",
+				Optional:            true,
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(1),
+				},
+			},
+			"password": schema.StringAttribute{
+				MarkdownDescription: "The password for authentication with the mail server.",
+				Optional:            true,
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(1),
+				},
+			},
+			"port": schema.Int64Attribute{
+				MarkdownDescription: "The port number of the mail server.",
+				Required:            true,
+			},
+			"subject_prefix": schema.StringAttribute{
+				MarkdownDescription: "A prefix to use for the subject of all outgoing mails.",
+				Optional:            true,
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(1),
+				},
+			},
+			"use_ssl": schema.BoolAttribute{
+				MarkdownDescription: "When set to 'true', uses a secure connection to the mail server.",
+				Optional:            true,
+				Computed:            true,
+				Default:             booldefault.StaticBool(false),
+			},
+			"use_tls": schema.BoolAttribute{
+				MarkdownDescription: "When set to 'true', uses Transport Layer Security when connecting to the mail server.",
+				Optional:            true,
+				Computed:            true,
+				Default:             booldefault.StaticBool(false),
+			},
+		},
+	}
+}
+
+func (r *MailServerResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+	r.ProviderData = req.ProviderData.(utilsdk.ProvderMetadata)
```

Comment:
slightly slimmer logic would be to flip this and put line 176 inside the condition